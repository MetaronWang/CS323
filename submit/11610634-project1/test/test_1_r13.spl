int global;
struct my_struct
{
  int code;
  char data;
};
int test_1_r02()
{
  struct my_struct obj;
  obj.code = global;
  for (; global<10 ; i=i+1){
    global = global + 1;
  }
}
