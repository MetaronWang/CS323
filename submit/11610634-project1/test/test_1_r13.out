Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      ExtDecList (1)
        VarDec (1)
          ID: global
      SEMI
    ExtDefList (2)
      ExtDef (2)
        Specifier (2)
          StructSpecifier (2)
            STRUCT
            ID: my_struct
            LC
            DefList (4)
              Def (4)
                Specifier (4)
                  TYPE: int
                DecList (4)
                  Dec (4)
                    VarDec (4)
                      ID: code
                SEMI
              DefList (5)
                Def (5)
                  Specifier (5)
                    TYPE: char
                  DecList (5)
                    Dec (5)
                      VarDec (5)
                        ID: data
                  SEMI
            RC
        SEMI
      ExtDefList (7)
        ExtDef (7)
          Specifier (7)
            TYPE: int
          FunDec (7)
            ID: test_1_r02
            LP
            RP
          CompSt (8)
            LC
            DefList (9)
              Def (9)
                Specifier (9)
                  StructSpecifier (9)
                    STRUCT
                    ID: my_struct
                DecList (9)
                  Dec (9)
                    VarDec (9)
                      ID: obj
                SEMI
            StmtList (10)
              Stmt (10)
                Exp (10)
                  Exp (10)
                    Exp (10)
                      ID: obj
                    DOT
                    ID: code
                  ASSIGN
                  Exp (10)
                    ID: global
                SEMI
              StmtList (11)
                Stmt (11)
                  FOR
                  LP
                  ForExps (11)
                    SEMI
                    Exp (11)
                      Exp (11)
                        ID: global
                      LT
                      Exp (11)
                        INT: 10
                    SEMI
                    Exp (11)
                      Exp (11)
                        ID: i
                      ASSIGN
                      Exp (11)
                        Exp (11)
                          ID: i
                        PLUS
                        Exp (11)
                          INT: 1
                  RP
                  Stmt (11)
                    CompSt (11)
                      LC
                      StmtList (12)
                        Stmt (12)
                          Exp (12)
                            Exp (12)
                              ID: global
                            ASSIGN
                            Exp (12)
                              Exp (12)
                                ID: global
                              PLUS
                              Exp (12)
                                INT: 1
                          SEMI
                      RC
            RC
